core/org.osate.ui/src/org/osate/ui/navigator/AadlElementContentProvider.java

```diff
@@ -84,17 +93,20 @@ public Object getParent(Object element) {
 	public boolean hasChildren(Object element) {
 		if (element instanceof IFile || element instanceof ContributedAadlStorage) {
 			return true;
-		} else if (element instanceof ComponentImplementation) {
-			return !((ComponentImplementation) element).getOwnedAnnexSubclauses().isEmpty();
 		} else if (!shouldExpand(element)) {
 			return false;
 		}
+
+		/*
+		 * XXX This is wrong for Classifiers because we filter out Generalization nodes when we get the children, but we
+		 * don't take that into account here.
+		 */
```

Comment:
This is confusing. I'd delete it.

Reply:
I rewrote the comment to better explain the situation.  I don't want to get rid of the comment entirely, because I'm trying to mark a place where I know I am not doing something completely correctly.